util/EFSOI_Utilities/src/gridinfo_efsoi.f90

```diff
@@ -0,0 +1,461 @@
+real(nemsio_realkind), dimension(nlons*nlats) :: nems_wrk
+type(sigio_data) sigdata
+type(sigio_head) sighead
+type(nemsio_gfile) :: gfile
+
+integer(i_kind) iret,i,j,nlonsin,nlatsin,u_ind,v_ind,tv_ind,q_ind,ps_ind
+integer, intent(in) :: nc3d
+integer, dimension(0:nc3d), intent(in) :: clevels
+
+
+iunit = 77
+kap = rd/cp
+kapr = cp/rd
+kap1 = kap + one
+nlevs_pres=nlevs+1
+
+
+if (nproc .eq. 0) then
+   filename = trim(adjustl(datapath))//trim(adjustl(fileprefix))//"ensmean"
+   print *,'reading file: ',trim(filename)
+   
+   if (use_gfs_nemsio) then
+       call nemsio_init(iret=iret)
+       if(iret/=0) then
+          write(6,*)'grdinfo: gfs model: problem with nemsio_init, iret=',iret, ', file: ', trim(filename)
+          call stop2(23)
+       end if
+       call nemsio_open(gfile,filename,'READ',iret=iret)
+       if (iret/=0) then
+          write(6,*)'grdinfo: gfs model: problem with nemsio_open, iret=',iret, ', file: ', trim(filename)
+          call stop2(23)
+       endif
+       call nemsio_getfilehead(gfile,iret=iret, dimx=nlonsin, dimy=nlatsin,&
+                               dimz=nlevsin,jcap=ntrunc,idvc=idvc)
+       ! set ntrunc to nlats if missing
+       ! (only used for inflation smoothing and mass balance adjustment if use_gfsnemsio = T)
+       ! FV3GFS write component does not include JCAP, infer from nlatsin
+       if (ntrunc < 0) ntrunc = nlatsin-2
+       if (iret/=0) then
+          write(6,*)'grdinfo: gfs model: problem with nemsio_getfilehead, iret=',iret, ', file: ', trim(filename)
+          call stop2(23)
+       endif
+       print *,'ntrunc = ',ntrunc
+       if (nlons /= nlonsin .or. nlats /= nlatsin .or. nlevs /= nlevsin) then
+         print *,'incorrect dims in nemsio file'
+         print *,'expected',nlons,nlats,nlevs
+         print *,'got',nlonsin,nlatsin,nlevsin
+         call stop2(23)
+       end if
+   else if (use_gfs_ncio) then
+	   
+      dset = open_dataset(filename)
+	  
+      londim = get_dim(dset,'grid_xt'); nlonsin = londim%len
+      latdim = get_dim(dset,'grid_yt'); nlatsin = latdim%len
+      levdim = get_dim(dset,'pfull');   nlevsin = levdim%len
+      idvc = 2; ntrunc = nlatsin-2
+	  
+      if (nlons /= nlonsin .or. nlats /= nlatsin .or. nlevs /= nlevsin) then
+         print *,'incorrect dims in netcdf file'
+         print *,'expected',nlons,nlats,nlevs
+         print *,'got',nlonsin,nlatsin,nlevsin
+         call stop2(23)
+      end if
+	  
+   else
+      ! define sighead on all tasks.
+      call sigio_sropen(iunit,trim(filename),iret)
+      if (iret /= 0) then
+         print *,'error reading file in gridinfo ',trim(filename),' on task',nproc
+         call stop2(24)
+      end if
+      call sigio_srhead(iunit,sighead,iret)
+      if (iret /= 0) then
+         print *,'error reading file in gridinfo ',trim(filename),' on task',nproc
+         call stop2(24)
+      end if
+      call sigio_sclose(iunit,iret)
+      ntrunc = sighead%jcap
+   endif
```

Comment:
we are not using spectral model output for a while.  This should be cleaned.

Reply:
gridinfo_efsoi.f90 removed, using enkf version